components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/action/PreIssueAccessTokenProcessor.java

```diff
@@ -0,0 +1,719 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.oauth.action;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.action.execution.ActionExecutionResponseProcessor;
+import org.wso2.carbon.identity.action.execution.model.ActionExecutionStatus;
+import org.wso2.carbon.identity.action.execution.model.ActionInvocationErrorResponse;
+import org.wso2.carbon.identity.action.execution.model.ActionInvocationSuccessResponse;
+import org.wso2.carbon.identity.action.execution.model.ActionType;
+import org.wso2.carbon.identity.action.execution.model.Event;
+import org.wso2.carbon.identity.action.execution.model.PerformableOperation;
+import org.wso2.carbon.identity.oauth.action.model.AccessToken;
+import org.wso2.carbon.identity.oauth.action.model.ClaimPathInfo;
+import org.wso2.carbon.identity.oauth.action.model.OperationExecutionResult;
+import org.wso2.carbon.identity.oauth.action.model.PreIssueAccessTokenEvent;
+import org.wso2.carbon.identity.oauth2.token.OAuthTokenReqMessageContext;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * This class is responsible for processing the response received from the action execution
+ * of the pre issue access token.
+ */
+public class PreIssueAccessTokenProcessor implements ActionExecutionResponseProcessor {
+
+    private static final Log LOG = LogFactory.getLog(PreIssueAccessTokenProcessor.class);
+    private static final String SCOPE_PATH_PREFIX = "/accessToken/scopes/";
+    private static final String CLAIMS_PATH_PREFIX = "/accessToken/claims/";
+    private static final Pattern NQCHAR_PATTERN = Pattern.compile("^[\\x21\\x23-\\x5B\\x5D-\\x7E]+$");
+    private static final Pattern STRING_OR_URI_PATTERN =
+            Pattern.compile("^([a-zA-Z][a-zA-Z0-9+.-]*://[^\\s/$.?#].\\S*)|(^[a-zA-Z0-9.-]+$)");
+    private static final String LAST_ELEMENT_CHARACTER = "-";
+    private static final char PATH_SEPARATOR = '/';
+
+    @Override
+    public ActionType getSupportedActionType() {
+
+        return ActionType.PRE_ISSUE_ACCESS_TOKEN;
+    }
+
+    @Override
+    public ActionExecutionStatus processSuccessResponse(ActionType actionType, Map<String, Object> eventContext,
+                                                        Event event,
+                                                        ActionInvocationSuccessResponse
+                                                                actionInvocationSuccessResponse) {
+
+        OAuthTokenReqMessageContext tokenMessageContext =
+                (OAuthTokenReqMessageContext) eventContext.get("tokenMessageContext");
```

Comment:
Add or reuse a constant for "tokenMessageContext"

Reply:
This is the only place this literal is used. I don't see a requirement to define as a constant. Can incorporate though